sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -47,364 +48,444 @@ const alwaysBeDeletedSubscription = "alwaysbedeletedsubscription";
-          should.equal(
-            response.createdAt,
-            undefined,
-            "Should receive empty result and just the raw response"
-          );
-        }
-        // Error is undefined for queues, topics - but not for non-existent subscriptions and rules
-      });
+      should.equal(Array.isArray(response), true, "Result must be any array for list requests");
+      should.equal(
+        response.length,
+        testCase.entityType == EntityType.RULE ? 2 : 1,
+        "Result must be an empty array"
+      );
+      should.equal(
+        response[0][testCase.entityType.toLowerCase() + "Name"],
+        testCase.entityType == EntityType.RULE
+          ? testCase.alwaysBeDeletedEntity
+          : testCase.alwaysBeExistingEntity,
+        "Entity name mismatch"
+      );
+    });
+
+    it(`Creates a non-existent ${testCase.entityType} entity successfully`, async () => {
+      let response;
+
+      try {
+        response = await deleteEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
+      response = await createEntity(
+        testCase.entityType,
+        testCase.alwaysBeExistingEntity,
+        testCase.parentTopicName,
+        testCase.parentSubscriptionName
+      );
+      should.equal(
+        response[testCase.entityType.toLowerCase() + "Name"],
+        testCase.alwaysBeExistingEntity,
+        "Entity name mismatch"
+      );
+    });
+
+    it(`Creating an existent ${testCase.entityType} entity throws an error`, async () => {
+      let response;
+
+      try {
+        await createEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        response = err;
+      }
+
+      should.equal(response.statusCode, 409, "Error must not be undefined");
+      should.equal(
+        response.message.startsWith("The messaging entity") ||
+          response.message.startsWith("Entity") ||
+          response.message.startsWith("SubCode") ||
+          response.message.startsWith("No service"),
+        true,
+        `Expected error message to be a textual content but got "${response.message}"`
+      );
     });
-  }
-);
+
+    it(`Lists available ${testCase.entityType} entities successfully`, async () => {
+      let response;
+
+      response = await listEntities(
+        testCase.entityType,
+        testCase.parentTopicName,
+        testCase.parentSubscriptionName
+      );
+
+      should.equal(Array.isArray(response), true, "Result must be any array for list requests");
+    });
+
+    it(`Updates an existent ${testCase.entityType} entity successfully`, async () => {
+      let response: any;
+
+      response = await updateEntity(
```

Comment:
How do we know the entity is already existing?

Reply:
Updated, all of these rely on the "alwaysBeExisting.."